core/shared/src/main/scala/clue/package.scala

```diff
@@ -1,13 +1,65 @@
-// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
 // For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
 import cats.syntax.all._
+import io.chrisdavenport.log4cats.Logger
 import scala.concurrent.duration.FiniteDuration
+import cats.MonadError
+import cats.effect.Concurrent
+import cats.effect.concurrent.Deferred
 
 package object clue {
   type CloseReason[CE]               = Either[Throwable, CE]
+  // Int = Attempt #. Will only be 0 immediately after a close.
+  // For first connection, it will be called the first time with 1, after 1st connection attempt.
   type ReconnectionStrategy[CE]      = (Int, CloseReason[CE]) => Option[FiniteDuration]
   type WebSocketReconnectionStrategy = ReconnectionStrategy[WebSocketCloseEvent]
+
+  type WebSocketClient[F[_], S] =
+    PersistentStreamingClient[F, S, WebSocketCloseParams, WebSocketCloseEvent]
+
+  type WebSocketBackend[F[_]]    = PersistentBackend[F, WebSocketCloseParams, WebSocketCloseEvent]
+  type WebSocketConnection[F[_]] = PersistentConnection[F, WebSocketCloseParams]
+
+  type ApolloWebSocketClient[F[_], S] =
+    ApolloClient[F, S, WebSocketCloseParams, WebSocketCloseEvent]
+
+  protected[clue] type Latch[F[_]] = Deferred[F, Either[Throwable, Unit]]
+
+  final implicit class StringOps(val str: String) extends AnyVal {
```

Comment:
Maybe this is a general utility we could use in other places

Reply:
Sure, if other people like this syntax I'll be happy to move it (or copy it? I'm not sure I want to introduce non-general dependencies).